torch/_inductor/scheduler.py

```diff
@@ -1117,23 +1117,24 @@ def codegen(self):
                     or node.is_template()
                 ):
                     self.flush()
+                    print(f"Switching from {self.current_device} to {device}")
                     self.current_device = device
+            with torch.cuda.device(self.current_device):
```

Comment:
I don't think you should be setting current_device here, instead you should be emitting this code in the codegen_wrapper (and this unfortunately would require threading the initial indent through all IndentedBuffers)

Reply:
ok- will try to do that. thanks!

Reply:
removing this for now and will intercept https://github.com/pytorch/pytorch/pull/90934